embedded-graphics/src/draw_target.rs

```diff
@@ -0,0 +1,304 @@
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// /// A fake display 64px x 64px where each pixel is stored as a single `u8`
+/// struct FastExampleDisplay {
+///     framebuffer: [u8; 64 * 64],
+///     iface: SPI1,
+/// }
+///
+/// impl FastExampleDisplay {
+///     /// Send buffer to the display
+///     pub fn flush(&self) -> Result<(), ()> {
+///         self.iface.send_bytes(&self.framebuffer)
+///     }
+///
+///     /// A HW-accelerated method for drawing rectangles
+///     pub fn fast_rectangle(&self, rect: &Styled<Rectangle, PrimitiveStyle<Gray8>>) {
+///         // Does some speedy drawing
+///     }
+/// }
+///
+/// impl DrawTarget<Gray8> for FastExampleDisplay {
+///     type Error = core::convert::Infallible;
+///
+///     /// Draw a `pixel` that has a color defined as `Gray8`
+///     fn draw_pixel(&mut self, pixel: Pixel<Gray8>) -> Result<(), Self::Error> {
+///         let Pixel(coord, color) = pixel;
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord.x + coord.y * 64;
+///         self.framebuffer[index as usize] = color.luma();
+///
+///         Ok(())
+///     }
+///
+///     fn size(&self) -> Size {
+///         Size::new(64, 64)
+///     }
+///
+///     /// Use the accelerated method when drawing rectangles
+///     fn draw_rectangle(
+///         &mut self,
+///         item: &Styled<Rectangle, PrimitiveStyle<Gray8>>,
+///     ) -> Result<(), Self::Error> {
+///         self.fast_rectangle(item);
+///
+///         Ok(())
+///     }
+/// }
+///
+/// let mut display = FastExampleDisplay {
+///     framebuffer: [0; 4096],
+///     iface: SPI1,
+/// };
+///
+/// // Draw a rectangle from (10, 20) to (30, 40) with a white stroke
+/// let rect = egrectangle!(
+///     top_left = (10, 20),
+///     bottom_right = (30, 40),
+///     style = primitive_style!(stroke_color = Gray8::WHITE)
+/// );
+/// rect.draw(&mut display)?; // Uses the accelerated draw_rectangle function
+///
+/// // Update the display
+/// display.flush().expect("Failed to send data to display");
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// [`Drawable`]: ../drawable/trait.Drawable.html
+/// [`Pixel`]: ../drawable/struct.Pixel.html
+/// [`draw_pixel`]: ./trait.DrawTarget.html#method.draw_pixel
+/// [`DrawTarget`]: ./trait.DrawTarget.html
+pub trait DrawTarget<C>
+where
+    C: PixelColor,
+{
+    /// Error type to return when a draw or display change operation fails.
+    ///
+    /// This error is returned when a drawing operation fails, or when an error updating/clearing
+    /// the display occurs. The examples in this crate use `core::convert::Infallible`, however a
+    /// more descriptive error type should be used with real hardware.
+    type Error;
+
+    /// Draws a pixel on the display.
+    ///
+    /// Note that some displays require a "flush" operation to actually write changes to the
+    /// framebuffer.
```

Comment:
This should be moved to trait-level docs, because it applies to all `draw_...` methods. And it should also mention that e-g don't provide such a method and that the driver docs probably provide more information.